tools/building/createTest.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+import pathlib
+
+
+def is_precice_root(dir):
+    detect = [
+        "CHANGELOG.md", "CMakeLists.txt", "LICENSE", "src", "tests", "cmake"
+    ]
+    return all(map(lambda c: os.path.exists(os.path.join(dir, c)), detect))
+
+
+def find_precice_root():
+    search_depth = 4
+    current = pathlib.Path(os.path.curdir).absolute()
+    candidats = [current] + list(current.parents)[:search_depth]
+    for dir in candidats:
+        if is_precice_root(dir):
+            return dir
+    raise "Unable to find the root directory of precice"
+
+
+def patharg(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given suite name \"{}\" must use CamelCase.".format(arg))
+    return arg
+
+
+def testname(arg):
+    assert (arg)
+    if " " in arg:
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" cannot contain spaces.".format(arg))
+    if "." in arg:
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" cannot contain the file extensions.".
+            format(arg))
+    if not arg[0].isupper():
+        raise argparse.ArgumentTypeError(
+            "The given test name \"{}\" must use CamelCase.".format(arg))
+    return arg
+
+
+PRECICE_TEST_BODY = """
+{
+  PRECICE_TEST(TODO);
```

Comment:
```suggestion
  PRECICE_TEST(false);
```
To ensure that this is actually implemented and the test fails when only the skeleton is used (also a good first check: It my newly created test executed?).

Reply:
This is the test scenario, not a check :wink:

Reply:
Right :sweat_smile: I meant something in this direction:

```suggestion
  PRECICE_TEST(TODO);
  
  // Implement your test here.
  BOOST_TEST(false);
```